packages/devtools/src/schema/compiler.ts

```
@@ -0,0 +1,315 @@
+    }
+
+    console.log('_compileEmbedObject create', name, definition)
+    const object: CompileObject = {
+      type: 'object',
+      title: name,
+      properties: {},
+      additionalProperties: false,
+    }
+    const required: Array<string> = []
+    let refs: Array<string> = []
+
+    for (const [key, field] of Object.entries(definition)) {
+      if (field.required) {
+        required.push(key)
+      }
+
+      console.log('compile field', key, field)
+
+      let value: SchemaWithRefs | void
+      switch (field.type) {
+        case 'enum':
+          // TODO
+          break
+        case 'list':
+          value = this._compileList(name, key, field)
+          break
+        case 'object':
+          value = this._compileObjectReference(name, key, field)
+          break
+        case 'scalar':
+          value = this._compileScalar(field)
+          break
+        case 'union':
+          // TODO
+          break
+        case 'view':
+          throw new Error(
+            `Unsupported view on field ${key} of object ${name}. Views can only be set on models.`
+          )
+      }
+
+      if (value == null) {
+        throw new Error(`Could not compile value for field ${key} of object ${name}`)
+      }
+      object.properties[key] = value.schema
+      refs = [...refs, ...value.refs]
+    }
+
+    if (required.length !== 0) {
+      object.required = required
+    }
+
+    const schemaWithRefs = { schema: object, refs }
+    this.#refs[name] = schemaWithRefs
+
+    return schemaWithRefs
+  }
+
+  _compileList(
+    objectName: string,
+    fieldName: string,
+    definition: ListFieldDefinition
+  ): SchemaWithRefs {
+    const list: JSONSchema.Array = { type: 'array', maxItems: definition.maxLength }
+    if (definition.minLength != null) {
+      list.minItems = definition.minLength
+    }
+
+    let item: SchemaWithRefs | void
+    switch (definition.item.type) {
+      case 'enum':
+        // TODO
```

I know this is only temporary, but can we throw here just as good practice to make sure that we don't end up with silently failing functions in the future :) ?

---

Are you looking at the latest version of the changes? That's not what's at this line: https://github.com/ceramicstudio/js-composedb/pull/10/files#diff-9f8daf907f196e3ea70a4159f63a7f187388e32d4e56a695e3f55241859b4749R175